read-fonts/src/tables/postscript/font.rs

```diff
@@ -0,0 +1,242 @@
+pub struct Font<'a> {
+    /// Root table data for accessing items by offset.
+    pub table_data: &'a [u8],
+    // TODO: add type for font matrix
+    pub matrix: Option<[Fixed; 6]>,
+    pub charstrings: Option<Index<'a>>,
+    pub fd_array: Option<Index<'a>>,
+    pub fd_select: Option<FdSelect<'a>>,
+    pub global_subrs: Index<'a>,
+    pub units_per_em: u16,
+    pub kind: FontKind<'a>,
+}
+
+impl<'a> Font<'a> {
+    pub fn new(font: &impl TableProvider<'a>) -> Result<Self, Error> {
+        let units_per_em = font.head()?.units_per_em();
+        if let Ok(cff2) = font.cff2() {
+            Self::from_cff2(&cff2, units_per_em)
+        } else {
+            Self::from_cff(&font.cff()?, units_per_em)
+        }
+    }
+
+    pub fn from_cff(cff: &Cff<'a>, units_per_em: u16) -> Result<Self, Error> {
+        let table_data = cff.offset_data().as_bytes();
+        let mut data = FontData::new(cff.trailing_data());
+        let names = Index1::read(data)?;
+        data = data
+            .split_off(names.size_in_bytes()?)
+            .ok_or(ReadError::OutOfBounds)?;
+        let top_dicts = Index1::read(data)?;
+        data = data
+            .split_off(top_dicts.size_in_bytes()?)
+            .ok_or(ReadError::OutOfBounds)?;
+        let strings = Index1::read(data)?;
+        data = data
+            .split_off(strings.size_in_bytes()?)
+            .ok_or(ReadError::OutOfBounds)?;
+        let global_subrs = Index::new(data.as_bytes(), false)?;
+        let top_dict_data = top_dicts.get(0)?;
+        let top_items = TopDictItems::new(table_data, top_dict_data, false)?;
+        let private_dict_range = top_items.private_dict_range.clone();
+        Ok(Self::new_impl(
+            table_data,
+            units_per_em,
+            top_items,
+            global_subrs,
+            FontKind::Cff(CffFont {
+                names,
+                top_dicts,
+                strings,
+                private_dict_range,
+            }),
+        ))
+    }
+
+    pub fn from_cff2(cff2: &Cff2<'a>, units_per_em: u16) -> Result<Self, Error> {
+        let top_dict_data = cff2.top_dict_data();
+        let global_subrs_data = cff2.trailing_data();
+        let global_subrs = Index::new(global_subrs_data, true)?;
+        let table_data = cff2.offset_data().as_bytes();
+        let top_items = TopDictItems::new(table_data, top_dict_data, true)?;
+        let var_store = top_items.var_store.clone();
+        Ok(Self::new_impl(
+            table_data,
+            units_per_em,
+            top_items,
+            global_subrs,
+            FontKind::Cff2(Cff2Font { var_store }),
+        ))
+    }
+
+    pub fn string(&self, id: StringId) -> Option<Latin1String<'a>> {
+        Some(match id.standard_string() {
+            Ok(name) => name,
+            Err(ix) => match &self.kind {
+                FontKind::Cff(cff) => Latin1String::new(cff.strings.get(ix).ok()?),
+                _ => return None,
+            },
+        })
```

Comment:
It's a habit I've fallen into that is probably not appropriate here because of the small number of match arms. I generally do it when there are a bunch of arms all returning `Some(...)` or `Ok(...)` and I find the repetition really distracting so I factor it out.

Your suggestion LGTM!